### ISSUE TITLE
MA0102 triggers on constructor

### ISSUE THREAD

**Issue by RobinHood70** [meziantou/Meziantou.Analyzer#236]
**Context**
Rule identifier: _MA0102_
Target Framework: _.NET 5.0_
Language version: _9.0_

**Describe the bug**
If a structure provides a constructor, MA0102 suggests that it should be made read-only, which is impossible.

**To Reproduce**
Code or steps to reproduce the behavior:

````c#
	public struct PauseToken : IEquatable<PauseToken>
	{
		// MA0102 triggers here
		public PauseToken(PauseTokenSource source) => this.source = source;
	}
````

**Comment by meziantou** [comment]
Thanks for reporting this issue!

The version 1.0.656 should fix the issue (available in a few minutes)
